### PR TITLE
feat(rds): add SQL Server version 15.00.4043.16.v1

### LINKED ISSUE #10273: [rds] Support for RDS SQL Version 15.00.4043.16.v1
###Description

At present the latest RDS Microsoft SQL server version (15.00.4043.16.v1) is not supported by CDK

### Use Case

Needed to create a Microsoft SQL DB RDS instance, option group and parameter using version 15.00.4043.16.v1

### Proposed Solution
Following change need to be done in instance-engine.ts

/** Version "15.00" (only a major version, without a specific minor version). */
  public static readonly VER_15 = SqlServerEngineVersion.of('15.00', '15.00');
  /** Version "15.00.4043.16.v1". */
  public static readonly VER_15_00_4043_16_V1 = SqlServerEngineVersion.of('15.00.4043.16.v1', '15.00');

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by abjoshi2785**
Closes #10273

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by shivlaks — DISMISSED** [review]

File: packages/@aws-cdk/aws-rds/lib/instance-engine.ts [hunk 1]
```diff
@@ -875,6 +875,11 @@ export class SqlServerEngineVersion {
   public static readonly VER_14_00_3049_1_V1 = SqlServerEngineVersion.of('14.00.3049.1.v1', '14.00');
   /** Version "14.00.3192.2.v1". */
   public static readonly VER_14_00_3192_2_V1 = SqlServerEngineVersion.of('14.00.3192.2.v1', '14.00');
+
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/lib/instance-engine.ts:878]:
```suggestion

```

Comment (shivlaks) [reply packages/@aws-cdk/aws-rds/lib/instance-engine.ts]: thanks @njlynch !

File: packages/@aws-cdk/aws-rds/lib/instance-engine.ts [hunk 2]
```diff
@@ -875,6 +875,10 @@ export class SqlServerEngineVersion {
   public static readonly VER_14_00_3049_1_V1 = SqlServerEngineVersion.of('14.00.3049.1.v1', '14.00');
   /** Version "14.00.3192.2.v1". */
   public static readonly VER_14_00_3192_2_V1 = SqlServerEngineVersion.of('14.00.3192.2.v1', '14.00');
+  /** Version "15.00" (only a major version, without a specific minor version). */
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/instance-engine.ts:878]: Please add a newline before this line. Just without any trailing whitespace :).

**Review by skinny85 — DISMISSED** [review]
Thanks for the PR @abjoshi2785 !

File: packages/@aws-cdk/aws-rds/lib/instance-engine.ts (hunk 2 above)
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/lib/instance-engine.ts:878]:
```suggestion

  /** Version "15.00" (only a major version, without a specific minor version). */
```

**Review by skinny85 — DISMISSED** [review]

**Review by njlynch — DISMISSED** [review]

**Comment by skinny85** [comment]
@abjoshi2785 did you ["Allow changes to be pushed to your fork"](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)? I think we can't merge your PR until you do...

**Review by skinny85 — CHANGES_REQUESTED** [review]
Putting in "Request changes" to try and unblock the Mergify queue.

**Review by skinny85 — APPROVED** [review]
Seems the queue is unclogged, approving again 🙂

**Comment by abjoshi2785** [comment]
Thank you! :)

**Comment by skinny85** [comment]
> Thank you! :)

No, thank _you_ for the contribution 🙂

**Pull request merged**